encrypt-config/FlowConfigEncryptor.cpp

```diff
@@ -0,0 +1,174 @@
+  switch (type) {
+    case Type::Processor: return "Processor";
+    case Type::ControllerService: return "Controller service";
+  }
+  return invalid_tag;
+}
+}  // namespace magic_enum::customize
+
+namespace {
+std::vector<SensitiveProperty> listSensitiveProperties(const minifi::core::ProcessGroup &process_group) {
+  std::vector<SensitiveProperty> sensitive_properties;
+
+  std::vector<minifi::core::Processor *> processors;
+  process_group.getAllProcessors(processors);
+  for (const auto *processor : processors) {
+    gsl_Expects(processor);
+    for (const auto& [_, property] : processor->getProperties()) {
+      if (property.isSensitive()) {
+        sensitive_properties.push_back(SensitiveProperty{
+            .type = Type::Processor,
+            .component_id = processor->getUUID(),
+            .component_name = processor->getName(),
+            .property_name = property.getName(),
+            .property_display_name = property.getDisplayName()});
+      }
+    }
+  }
+
+  for (const auto* controller_service_node : process_group.getAllControllerServices()) {
+    gsl_Expects(controller_service_node);
+    const auto* controller_service = controller_service_node->getControllerServiceImplementation();
+    gsl_Expects(controller_service);
+    for (const auto& [_, property] : controller_service->getProperties()) {
+      if (property.isSensitive()) {
+        sensitive_properties.push_back(SensitiveProperty{
+            .type = Type::ControllerService,
+            .component_id = controller_service->getUUID(),
+            .component_name = controller_service->getName(),
+            .property_name = property.getName(),
+            .property_display_name = property.getDisplayName()});
+      }
+    }
+  }
+
+  return sensitive_properties;
+}
+
+template<typename Func>
+void encryptSensitiveValuesInFlowConfigImpl(
+    const minifi::encrypt_config::EncryptionKeys& keys, const std::filesystem::path& minifi_home, const std::filesystem::path& flow_config_path, Func create_overrides) {
+  const auto configure = std::make_shared<minifi::Configure>();
+  configure->setHome(minifi_home);
+  configure->loadConfigureFile(DEFAULT_NIFI_PROPERTIES_FILE);
+
+  bool encrypt_whole_flow_config_file = (configure->get(minifi::Configure::nifi_flow_configuration_encrypt) | minifi::utils::andThen(minifi::utils::string::toBool)).value_or(false);
+  auto encryptor = encrypt_whole_flow_config_file ? minifi::utils::crypto::EncryptionProvider::create(minifi_home) : std::nullopt;
+  auto filesystem = std::make_shared<minifi::utils::file::FileSystem>(encrypt_whole_flow_config_file, encryptor);
+
+  minifi::core::extension::ExtensionManager::get().initialize(configure);
+
+  minifi::core::flow::AdaptiveConfiguration adaptive_configuration{minifi::core::ConfigurationContext{
+      .flow_file_repo = minifi::core::createRepository("flowfilerepository"),
+      .content_repo = std::make_shared<minifi::core::repository::VolatileContentRepository>(),
```

Comment:
Would this work with passing null or NoopRepository? I don't think we need repositories for modifying the config.

Reply:
Good point; I've removed the repository creation in 0506380be319a185920c571ee0737556d86c89bd.